snapcraft/lifecycle.py

```diff
@@ -36,6 +40,21 @@ def _local_plugindir():
     return os.path.abspath(os.path.join('parts', 'plugins'))
 
 
+def execute(stage, part_names=None):
+    config = snapcraft.yaml.load_config()
+    repo.install_build_packages(config.build_tools)
+
+    if part_names:
+        config.validate_parts(part_names)
+        parts = {p for p in config.all_parts if p.name in part_names}
+    else:
+        parts = config.all_parts
+
+    for part in parts:
+        common.env = config.build_env_for_part(part)
+        part.pull()
```

Comment:
I'm confused here. It's doing only pull because you haven't migrated the other commands yet?